### PR TITLE
eos start generates a basic server

### PR DESCRIPTION
**Pull request by paul-codes-1**
I integrated a createServer action into the start command.  It's a really basic server right now, but it'd be helpful to pull in this change now to reduce future conflicts when we have to merge stuff in.

### CODE REVIEW

**Comment by paul-codes-1** [comment]
I also implemented the command eos server to launch a server on localhost:3000

**Comment by paul-codes-1** [comment]
Ok, I swtiched to ejs views. they're like erb but with javascript. 

if you run `eos start` then cd into the project, then run `eos server`, running webpack and naviagating to localhost:3000 gives you the welcome to your new eos app message.

**Comment by paul-codes-1** [comment]
Just noticed there are 2 webpack config files in templates and templates/start. Do we need both? Confused me for a minute.

**Comment by travisoneill** [comment]
I think that this should not be a default action for `eos start` since users will want to create frontend only projects with this framework.  I suggest an option to add as an argument `eos start myproject --backend express` this will also allow addition of other default servers in other frameworks.

**Comment by travisoneill** [comment]
I also think that the default server should be lightweight to avoid unnecessary complexity and dependencies that users may not need, and more like a basic 'hello-world' server:

``` JavaScript
const Express = require('express');
const app = Express();

app.get('/', function(req, res){
  console.log("Hello.  Node Server is running");
  res.send("Hello.  Node Server is running");
});

if (module === require.main) {
  var server = app.listen(process.env.PORT || 8000, function () {
    var port = server.address().port;
    console.log('Node Server listening on port %s', port);
  });
}

module.exports = server;
```

**Comment by paul-codes-1** [comment]
Good call. I changed it up so the server is not included by default. you must do `eos start backend` to generate the server now.

**Comment by alexmattson** [comment]
@paulmoliva 

Im on the other side. The server should definitely be included by default. The structure or the framework is to make starting a redux app as simple as possible which would include a backend. Although the front end only and rails options (as well as other backends) should be be present but only when specified. 

After thinking about it more I believe the backend options should be housed as a subset of start rather than there own function. The backend command should be run only once at the start. As such we shouldn't give the user access to be able to run the backend generation options independently of the start generation. 

Somthing like:

``` Javascript
  .action(function (cmd, env1, env2, env3) {
    // Start
    if (['start', 's'].includes(cmd)) {
      Actions.start(env1);
      if (env2 === 'backend') {
        Actions.backend(env3)
      } else {
        Actions.backend('express')
      }
    }
```

This can then lead to a switch statement on in the backend generation function that handles the logic of what to generate. 

In this case a custom backend would look like this:

```
eos start [name] backend rails 
```

or 

```
eos start [name] backend none
```

but by just using the `eos start [name]` you get a express backend like your making. This ensures that the the start command is making a fully functional app skeleton with a single command, which is the goal. 

Also we need to decouple the start generation from the beckend generation. Any files that have to be beckend specific (webpack, packages, ect.) should be handled in the backend function and removed from the basic start generation. Likewise backend shouldn't have to repeat any start file generation commands, which is how you currently have it structured. 

And to answer your other question, the second webpack is there to just have the reference path for a rails app. It is mostly a remnant from when this was generating a rails frontend.

**Comment by paul-codes-1** [comment]
Roger. Let's chat about it in the morning. I structured the start and backend generation the way I did because `express myapp` creates the project folder with the server, so they would overwrite each other if called separately. A possible fix is to just never call the `express` command and instead make templates to mimic what `express` creates without calling it directly.

**Comment by travisoneill** [comment]
Yeah.  There's no reason it couldn't have a default.  I think there should be a separate `eos g server [framewok] [name]` command that could be run with express when `eos start` is run.  This would also allow users to generate multiple servers if needed in their project by running `eos g server` as many times as needed.  Would also allow additional functionality to be added through a config option like `eos g server express [name] --config static json` for example.  

100% with you on the encapsulation.  Once you have this functionality up I'll write some defaults for a couple other backend frameworks.

File: templates/backend/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node ./bin/www"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "babel-core": "^6.13.2",
```
Comment (travisoneill) [inline templates/backend/package.json:17]: It looks like this is a package json for the express backend only.  If so, should remove the front end related dependencies.

**Pull request merged**